### DISCUSSION TITLE
Activity Level

### DISCUSSION THREAD

**Discussion by bobl61** [Drek282/PsychoStats#81] [Q&A]
I don't see any settings for player activity. Several players who haven't been on the server for over a month are showing 86% activity. Is this normal, or am I missing something?
![2023-05-30_110256](https://github.com/Drek282/PsychoStats/assets/107001969/e0243ab2-36c3-4426-b010-0b9a641f5356)

**Comment by Drek282** [comment]
See "Max inactivity" in  the main config settings.  I'm not 100% sure exactly how that works but you could try setting that lower to see if it affects the activity level.  However, you will have to erase and regenerate the stats if you do that.  "Max days exclusive" may also affect that rating.

@Rosenstein might understand how that works better than  I do.  I will say though that it appears to be working correctly.  I don't see a problem in the demos I am hosting.

**Reply by bobl61** [reply]
I have Max player inactivity (days) set to 5 (Recommended: 5 - 10 days). 
Max days is exclusive looks like it only has to do with kills (For example: If you have 1000 kills in the last 60 days and 'max days' is set to 30 with this enabled you will lose all kills past 30 days. If disabled you retain all the old kills).

**Comment by Drek282** [comment]
What is Max days set to?

Also try setting Max inactivity to the same as your Max days, then reset and regenerate the stats.

**Comment by Drek282** [comment]
The other question I have is, how often is your server up and how often do you run the stats script?

**Comment by Drek282** [comment]
Another question I have is, have you combined logs from two different servers?  It seems to me you did that before, and it caused some issues.  If you have done that, try resetting the stats and running the script again.

**Comment by Rosenstein** [comment]
![](https://i.imgur.com/WhdXocr_d.webp?maxwidth=760&fidelity=grand)
Activity, skill and rank has dropped for that player.

**Comment by bobl61** [comment]
I reset the stats. Seems okay now. I only have one server. Stats are updated twice a day (9 AM and 9PM). Server runs 24/7 (only down for maintenance).

**Comment by Drek282** [comment]
I know that PS isn't perfect, and sometimes things can get a little flaky with it.  But those kinds of issues aren't easy to diagnose and fix because they aren't consistent and repeatable.  It's extremely difficult to fix a problem you can't consistently replicate.

**Reply by bobl61** [reply]
Agreed. Nothing is perfect. I personally appreciate all the work you guys have done in trying to update PS. On a side note, does the country flags issue (a lot of players country code not being detected) have to do with [GeoIP](https://github.com/Drek282/PsychoStats/issues/65) ?

**Reply by Rosenstein** [reply]
I've noticed that, but that area is not fully updated yet.
What needs to be done still is to update list of countries so that old system works properly before upgrading to GeoIP2. 
I've updated IP ranges recently https://github.com/Drek282/PsychoStats/commit/c9a0ffd8e84b0a615272e54298f4382ce106da1e , check table 'ps_geoip_ip' in PHPMyAdmin to see if you have this many rows:

![image](https://github.com/Drek282/PsychoStats/assets/4959837/0334ecca-4b01-4a1f-97ca-8572834d0557)

**Comment by Drek282** [comment]
On Debian you need to install the package 'geoip-database'.  On Windows I'm not sure exactly what you need to do anymore, but what you are looking for is the file 'GeoIP.dat'

Then you go to the Google Map section of the main config in the Admin CP and enter the location into 'IP Lookup URL or Maxmind DB'.  On my system that is, '/usr/share/GeoIP/GeoIP.dat'.  The default XML still hosted on psychostats.com is decades out of date.

Note also that the IP Lookup only works for IPv4 addresses.

**Comment by Drek282** [comment]
@Rosenstein I'm not sure that the GeoIP lookup from the database is actually functional, and if it is, I don't know what needs to be done to make sure that is what is being used.  Do you leave the 'IP Lookup URL or Maxmind DB' field blank?

**Reply by Rosenstein** [reply]
I did not change that field from the default defunct psychostats url.
But I noticed that if you drop table 'ps_geoip_ip' player flags are removed from main page

**Comment by bobl61** [comment]
![2023-06-02_124026](https://github.com/Drek282/PsychoStats/assets/107001969/e1592926-7574-421f-ad5b-b6303d7f9def)
![2023-06-02_125712](https://github.com/Drek282/PsychoStats/assets/107001969/19995921-7182-492d-bd48-d2e65f05e43e)
![2023-06-02_125541](https://github.com/Drek282/PsychoStats/assets/107001969/60e20a40-1fba-4689-8f6d-25001cc3e67a)

**Reply by Rosenstein** [reply]
You have outdated table. Here's what you need to do:
While you have 'ps_geoip_ip' open in PMA, select drop command and drop that table, Or input 'DROP TABLE `ps_geoip_ip`;' in console. After that while still in the same table, switch to import tab and import https://github.com/Drek282/PsychoStats/blob/main/www/install/mysql/maxmind.sql

**Reply by bobl61** [reply]
Done.

**Reply by bobl61** [reply]
![2023-06-02_144247](https://github.com/Drek282/PsychoStats/assets/107001969/6d453969-9088-4d66-980a-7787877d6ecf)

**Reply by bobl61** [reply]
Worked a treat! Reset the stats. Now ALL player flags are correct :-)

**Comment by Drek282** [comment]
@Rosenstein Thanks.  I haven't looked at the code that deals with GeoIP and flags.  That's interesting.
